source/common/http/http2/codec_impl.h

```diff
@@ -379,6 +379,12 @@ class ConnectionImpl : public virtual Connection,
       // as the stream had pending data to process and the stream was not reset.
       bool buffered_on_stream_close_{false};
 
+      // Segment size for processing body data. Defaults to the value of high
+      // watermark of the *pending_recv_data_* buffer.
+      // If 0, we will process all buffered data.
+      uint32_t defer_processing_segment_size_{0};
```

Comment:
hm, if we're doing limited writes to encourage fair sharing I'd be inclined to instead cap this at the limit we'll read per connection on EPOLL loop which I thought was much smaller.  thoughts?